packages/ai-ide/src/browser/coder-agent.ts

```diff
@@ -38,5 +41,25 @@ export class CoderAgent extends AbstractStreamParsingChatAgent {
     override promptTemplates = [getCoderReplacePromptTemplate(true), getCoderReplacePromptTemplate(false)];
     override functions = [GET_WORKSPACE_DIRECTORY_STRUCTURE_FUNCTION_ID, GET_WORKSPACE_FILE_LIST_FUNCTION_ID, FILE_CONTENT_FUNCTION_ID, WriteChangeToFileProvider.ID];
     protected override systemPromptId: string | undefined = CODER_REPLACE_PROMPT_TEMPLATE_ID;
-
+    override async invoke(request: MutableChatRequestModel): Promise<void> {
+        await super.invoke(request);
+        this.suggest(request);
+    }
+    async suggest(context: ChatSession | ChatRequestModel): Promise<void> {
+        const model = ChatRequestModel.is(context) ? context.session : context.model;
+        const session = this.chatService.getSessions().find(candidate => candidate.model.id === model.id);
+        if (!(model instanceof MutableChatModel) || !session) { return; }
+        if (model.isEmpty()) {
+            model.setSuggestions([
```

Comment:
How do you test this, i.e. get to this state? If I write any message, the model is not empty anymore, correct?

Reply:
We can definitily also move this to a follow up, i.e. the ability to show suggestions early